libopae/api-shell.c

```diff
@@ -123,36 +124,33 @@ fpga_result fpgaGetPropertiesFromHandle(fpga_handle handle,
 	ASSERT_NOT_NULL_RESULT(
 		wrapped_handle->adapter_table->fpgaGetPropertiesFromHandle,
 		FPGA_NOT_SUPPORTED);
-	ASSERT_NOT_NULL_RESULT(
-		wrapped_handle->adapter_table->fpgaDestroyProperties,
-		FPGA_NOT_SUPPORTED);
 
 	res = wrapped_handle->adapter_table->fpgaGetPropertiesFromHandle(
-		wrapped_handle->opae_handle, &pr);
-
-	ASSERT_RESULT(res);
+		wrapped_handle->opae_handle, prop);
 
-	wrapped_properties = opae_allocate_wrapped_properties(
-		pr, wrapped_handle->adapter_table);
+	// If the output properties has a parent token set,
+	// then it will be a raw token. We need to wrap it.
+	if (fpgaPropertiesGetParent(*prop, &parent) == FPGA_OK) {
+		opae_wrapped_token *wrapped_parent =
+			opae_allocate_wrapped_token(
+				parent, wrapped_handle->adapter_table);
 
-	if (!wrapped_properties) {
-		OPAE_ERR("malloc failed");
-		res = FPGA_NO_MEMORY;
-		dres = wrapped_handle->adapter_table->fpgaDestroyProperties(
-			&pr);
+		if (!wrapped_parent) {
+			OPAE_ERR("malloc failed");
+			res = FPGA_NO_MEMORY;
+		} else {
+			sres = fpgaPropertiesSetParent(*prop, wrapped_parent);
+		}
 	}
 
-	*prop = wrapped_properties;
-
-	return res != FPGA_OK ? res : dres;
+	return res != FPGA_OK ? res : sres;
```

Comment:
Uhh... this is a little confusing to me..  how do you feel about something like:
```C
res = wrapped_h->at->fpgaGetPropertiesFromHandle(...);
if (res) {
        return res;
}
if (fpgaPropertiesGetParent(*prop, &parent) == FPGA_OK) {
        opae_wrapped_token *wp = opae_allocate_wrapped_token(...);
        if (!wp) {
            OPAE_ERR(...);
            return FPGA_NO_MEMORY;
        }
        return fpgaPropertiesSetParent(*prop, wp);
}
return res;
```

Reply:
thanks!